ddl/index.go

```diff
@@ -1008,8 +1010,17 @@ type addIndexWorker struct {
 	distinctCheckFlags []bool
 }
 
-func newAddIndexWorker(sessCtx sessionctx.Context, id int, t table.PhysicalTable, indexInfo *model.IndexInfo, decodeColMap map[int64]decoder.Column, reorgInfo *reorgInfo, jc *JobContext) *addIndexWorker {
-	index := tables.NewIndex(t.GetPhysicalID(), t.Meta(), indexInfo)
+func newAddIndexWorker(sessCtx sessionctx.Context, id int, t table.PhysicalTable, decodeColMap map[int64]decoder.Column, reorgInfo *reorgInfo, jc *JobContext) *addIndexWorker {
+	if !bytes.Equal(reorgInfo.currElement.TypeKey, meta.IndexElementKey) {
+		return nil
+	}
```

Comment:
Why do we need this check?

Reply:
I think it will never happend in TiDB now.
But just to be on the safe side, `return nil` will panic directly if there is something wrong with currElement , instead of using columnID to find a wrong index.

Reply:
If you think it is unnecessary, I can remove it.

Reply:
If the panic is expected, please add a log to help identifying the problem :)

Reply:
Done